src/lib/macos-compat.c

```diff
@@ -0,0 +1,100 @@
+#include <fcntl.h>
+#include <pthread.h>
+#include <signal.h>
+#include <unistd.h>
+#include <time.h>
+
+#include "macos-compat.h"
+
+// Time code based on: https://github.com/ChisholmKyle/PosixMachTiming/tree/master/src
+static inline void time_diff (const struct timespec* in, struct timespec* out) 
+{
+  out->tv_sec = in->tv_sec - out->tv_sec;
+  out->tv_nsec = in->tv_nsec - out->tv_nsec;
+
+  if (out->tv_sec < 0) {
+    out->tv_sec = 0;
+    out->tv_nsec = 0;
+    return;
+  }
+
+  if (out->tv_nsec > 0) {
+    return;
+  }
+
+  if (out->tv_sec == 0) {
+    out->tv_sec = 0;
+    out->tv_nsec = 0;
+  } else {
+    out->tv_sec = out->tv_sec - 1;
+    out->tv_nsec = out->tv_nsec + 1000000000;
+  }
+}
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-parameter"
+int clock_nanosleep (int clockid, int flags, const struct timespec *request, struct timespec *remain)
```

Comment:
oh hey this also solves #1121

Reply:
Nice, I'll add it to the commit message then :)